### PR TITLE
feat(mvc): Improve the Required decorator

### PR DESCRIPTION
**Pull request by Romakita**
## Informations

Type | Status | Migration
---|---|---
Feature | Ready | No

****

## Description
- Required can be use on property
- add Allow decorator

Closes: #130, #35

## Todos

- [x] Tests
- [x] Coverage
- [x] Example
- [x] Documentation

### CODE REVIEW

File: src/swagger/class/OpenApiParamsBuilder.ts [hunk 1]
```diff
@@ -80,14 +81,17 @@ export class OpenApiParamsBuilder extends OpenApiPropertiesBuilder {
...
     private createBaseParameter(inType: string, param: ParamMetadata): BaseParameter {
         let baseParam: BaseParameter = {
             name: (inType === "body") ? "body" : <string>param.expression,
             in: inType,
             required: !!param.required,
             description: ""
         };
+
+        if (param.required) {
+            this._responses[400] = {description: "Bad request"};
```
Comment (lwallent) [inline src/swagger/class/OpenApiParamsBuilder.ts]: I would consider changing the description as 400 already says it is a bad request :-)
Why not say "Missing required parameter"?

File: src/swagger/class/OpenApiPropertiesBuilder.ts [hunk 2]
```diff
@@ -61,6 +62,10 @@ export class OpenApiPropertiesBuilder {
             schema.description = schema.description || model.store.get("description");
         }
 
+        if (model.required) {
+            this._responses[400] = {description: "Bad request"};
```
Comment (lwallent) [inline src/swagger/class/OpenApiPropertiesBuilder.ts]: Ditto :-)

File: src/swagger/class/OpenApiParamsBuilder.ts (hunk 1 above)
Comment (Romakita) [reply src/swagger/class/OpenApiParamsBuilder.ts]: I'm agree with the proposed description ;)

But 422 ? You have an idea ?

**Comment by Romakita** [comment]
Thanks @lwallent for the code review ;)

**Pull request merged**